application/src/test/java/org/thingsboard/server/service/state/DefaultDeviceStateServiceTest.java

```diff
@@ -125,4 +140,81 @@ private void process(Map<EntityKeyType, Map<String, TsValue>> latest, long defau
         Assert.assertEquals(5000L, deviceStateData.getState().getInactivityTimeout());
     }
 
+    @Test
+    public void givenIncreaseInactivityTimeoutAndThenStateIsActive() throws Exception {
+        TelemetrySubscriptionService telemetrySubscriptionService = Mockito.mock(TelemetrySubscriptionService.class);
+        ReflectionTestUtils.setField(service, "tsSubService", telemetrySubscriptionService);
+        ReflectionTestUtils.setField(service, "defaultStateCheckIntervalInSec", 60);
+        ReflectionTestUtils.setField(service, "defaultActivityStatsIntervalInSec", 60);
+        ReflectionTestUtils.setField(service, "defaultInactivityTimeoutMs", 1);
+        ReflectionTestUtils.setField(service, "initFetchPackSize", 10);
+
+        Mockito.when(entityQueryRepository.findEntityDataByQueryInternal(Mockito.any())).thenReturn(new PageData<>());
+
+        service.init();
+        var tenantId = new TenantId(UUID.randomUUID());
+        var tpi = TopicPartitionInfo.builder().myPartition(true).build();
+        Mockito.when(partitionService.resolve(ServiceType.TB_CORE, tenantId, deviceId)).thenReturn(tpi);
+
+        var deviceIdInfo = new DeviceIdInfo(tenantId.getId(), null, deviceId.getId());
+
+        Mockito.when(deviceService.findDeviceIdInfos(Mockito.any()))
+                .thenReturn(new PageData<>(List.of(deviceIdInfo), 0, 1, false));
+
+        Method method = AbstractPartitionBasedService.class.getDeclaredMethod("initStateFromDB", Set.class);
```

Comment:
may be just call public `service.onTbApplicationEvent(new PartitionChangeEvent(...));`